File: university/src/main/java/com/gutorov/university/api/program/ProgramRs.java
package com.gutorov.university.api.program;

import java.util.UUID;

public class ProgramRs {
    private UUID id;
    private String name;

    public ProgramRs() {
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: src/main/java/com/gutorov/university/api/program/ProgramRs.kt
package com.gutorov.university.api.program

import com.gutorov.university.entity.ProgramEntity
import java.util.*
import java.util.stream.StreamSupport

data class ProgramRs(
    var id: UUID? = null,
    var name: String? = null
) {
    companion object {
        @JvmStatic
        fun fromEntity(entity: ProgramEntity?): ProgramRs {
            return ProgramRs(entity?.id, entity?.name)
        }

        @JvmStatic
        fun fromEntityList(entities: List<ProgramEntity>): List<ProgramRs> {
            return StreamSupport.stream(entities.spliterator(), false)
                .map(ProgramRs::fromEntity)
                .toList();
        }
    }
}